lib/oidcstrategy.js

```diff
@@ -69,18 +68,22 @@ const NONCE_MAX_AMOUNT = 10;
 const NONCE_LIFE_TIME = 3600; // second
 
 function makeProfileObject(src, raw) {
+  var emails = [];
+  if (src.upn)
+    emails = [src.upn];
```

Comment:
You are right. It is always in email format. But I am not sure if we can use that as email of the user. I saw you updated the code to set emails: src.emails. What is src here? json for id token? Is emails a new property?